java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java

```diff
@@ -130,13 +128,15 @@ public void visitFile(@Nullable Tree parsedTree) {
     }
   }
 
-  private boolean isNotJavaLangOrSerializable() {
-    String[] path = getContext().peekSourceCode().getName().split(Pattern.quote(File.separator));
-    boolean isJavaLang = path.length > 3 && "java".equals(path[path.length - 3]) && "lang".equals(path[path.length - 2]);
-    boolean isJavaLangAnnotation = path.length > 4 && "Annotation.java".equals(path[path.length - 1]) && "java".equals(path[path.length - 4])
-      && "lang".equals(path[path.length - 3]) && "annotation".equals(path[path.length - 2]);
-    boolean isSerializable = path.length > 3 && "Serializable.java".equals(path[path.length - 1]) && "java".equals(path[path.length - 3]) && "io".equals(path[path.length - 2]);
-    return !(isJavaLang || isJavaLangAnnotation || isSerializable);
+  private boolean isNotJavaLangOrSerializable(String packageName) {
+    if(packageName.equals("java/lang")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "java/lang" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)